### ISSUE TITLE
Cannot build -- MacOS

### ISSUE THREAD

**Issue by jajcayn** [brandonckelly/carma_pack#13]
Dear Brandon,

I was to build your carma package and I have following problem:
I use Homebrew and installed boost and boost-python through homebrew [and checked it whether it's working and it is] and also armadillo through homebrew. Numpy I got from pip. I did set all paths accordingly, but the `setup.py` gave me following errors:
```
non-existing path in '.': '/usr/include/'
non-existing path in '.': '/usr/include/'
non-existing path in '.': 'examples'
running install
error: can't combine user with prefix, exec_prefix/home, or install_(plat)base
```
Then I peeked into the setup.py and found out you are including `/usr/include` path, which I do not have and figured out I don't need to include it [since all the headers needed I already included from boost, armadillo and numpy], so I commented out including those, still having this error:
```
non-existing path in '.': 'examples'
running install
error: can't combine user with prefix, exec_prefix/home, or install_(plat)base
```

That I have no idea why, but figured out, hopefully it's only the examples, so I do not really need them, so also I commented out the line including examples. Finally, I got the compiler to work, but yielding following error:
`error: Command "clang++ -fno-strict-aliasing -fno-common -dynamic -g -O2 -DNDEBUG -g -fwrapv -O3 -Wall -I/usr/local/lib/python2.7/site-packages/numpy/core/include -I/usr/local/Cellar/boost/1.65.1/include -I/usr/local/Cellar/armadillo/8.200.1/include -I/usr/local/lib/python2.7/site-packages/numpy/core/include -I/usr/local/lib/python2.7/site-packages/numpy/core/include -c ./carpack.cpp -o build/temp.macosx-10.12-x
86_64-2.7/carpack.o -MMD -MF build/temp.macosx-10.12-x86_64-2.7/carpack.o.d -O3 -std=c++11 -stdlib=libc++" failed with exit status 1`

[and I am attaching full stdout & stderr from the build command `python2.7 setup.py install --prefix=/usr/local/lib/python2.7/site-packages/`]

[error.log](https://github.com/brandonckelly/carma_pack/files/1421723/error.log)

Have any ideas how can I fix this? 
Thanks a lot!
n.

**Comment by brandonckelly** [comment]
Hi Nikola,

I am sorry you are having build problems, but unfortunately I am do not
know why you are getting this error. My best guess would be that the
compiler cannot find the C++ 11 libraries? I know that at least one person
had problems with the more recent versions of boost and had to use an older
version.

Sorry I can't be more helpful, good luck!
-Brandon

**Comment by jajcayn** [comment]
Hi Brandon,

I was stupid before, finally somehow solved the problem. First of all, I was using boost version 1.65, when I fell back to 1.59 I was able to run it.. Still got the problem, when I had two versions of boost installed [1.59 and 1.65 in separate dirs], and I set BOOST_DIR variable pointing to the 1.59 one, the boost-python library is still linking from the newer version. I didn't know how to workaround, so I uninstalled the newer version and only kept the 1.59... 
It is working like this, so I am closing the issue.
Thanks for a reply!
 Nikola
